utils/cas_version_gen

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+#
+# Copyright(c) 2012-2020 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+SOURCES_DIR="$(realpath ../)"
+VERSION_FILE="$SOURCES_DIR/.metadata/cas_version"
+SUBMODULES=(
+    "ocf"
+    "test/functional/test-framework"
+)
+
+. "$SOURCES_DIR/version"
+
+# Check if we're inside a git repository
+if [[ -d "$SOURCES_DIR/.git" ]] && which git &>/dev/null &&\
+        (cd "$SOURCES_DIR" && git rev-parse --is-inside-work-tree &>/dev/null); then
+    echo "Git tree available - creating version file."
+
+    CAS_VERSION_BUILD=$(cd "$SOURCES_DIR" && git log --merges --oneline | wc -l)
+    LAST_COMMIT_HASH=$(cd "$SOURCES_DIR" && git log -1 --pretty=format:%H)
+    LAST_COMMIT_HASH_ABBR=$(cd "$SOURCES_DIR" && git log -1 --pretty=format:%h)
+    LAST_COMMIT_DATE=$(cd "$SOURCES_DIR" && git log -1 --pretty=format:%ci |\
+                       sed "s/ /T/" | sed "s/ //" | sed "s/00$/:00/")
+    LAST_COMMIT_TIMESTAMP=$(cd "$SOURCES_DIR" && git log -1 --pretty=format:%ct)
+    for SUBMOD in ${SUBMODULES[@]}; do
+        LAST_SUB_COMMIT_HASHES+=($(cd "$SOURCES_DIR/$SUBMOD" && git log -1 --pretty=format:%H))
+        LAST_SUB_COMMIT_HASHES_ABBR+=($(cd "$SOURCES_DIR/$SUBMOD" && git log -1 --pretty=format:%h))
+    done
+
+    if [[ $(cd "$SOURCES_DIR" && git tag --points-at HEAD) ]]; then
+        CAS_RELEASE="release"
+    elif [[ $(cd "$SOURCES_DIR" && git log -1 --pretty=format:%H)\
+            == $(cd "$SOURCES_DIR" && git log -1 --merges --pretty=format:%H) ]]; then
+        CAS_RELEASE="master"
+    else
+        CAS_RELEASE="devel.${LAST_COMMIT_TIMESTAMP}"
+    fi
+
+    CAS_VERSION=$(printf "%02d.%02d.%01d.%04d-%s" $CAS_VERSION_MAIN $CAS_VERSION_MAJOR\
```

Comment:
You could save 2 digits for minor part of CAS version. Also I suggest more digits for release part. Now it's not needed, but in future number of commits could exceed 9999 records, but I think somebody else should decide.